lib/puppet_x/certs/provider/keystore.rb

```diff
@@ -11,7 +11,25 @@ def destroy
         end
 
         def exists?
-          File.exist?(store)
+          return false unless File.exist?(store)
+
+          begin
+            keytool(
+              '-list',
+              '-keystore', store,
+              '-storepass:file', resource[:password_file],
+            )
+          rescue Puppet::ExecutionFailure => e
+            if e.message.include?('java.security.UnrecoverableKeyException')
+              Puppet.debug("Invalid password for #{store}")
+              #delete_keystore
```

Comment:
Turns out this isn't needed, but it would break noop mode where it would make changes. I've chosen to add removal to `generate_keystore`.